elements/pfe-tabs/pfe-tab.ts

```diff
@@ -1,157 +1,32 @@
-import type { ColorTheme } from '@patternfly/pfe-core';
-
-import { LitElement, html } from 'lit';
-import { customElement, property, query } from 'lit/decorators.js';
-
-import { pfelement, bound, observed, initializer, colorContextConsumer } from '@patternfly/pfe-core/decorators.js';
+import { customElement, property } from 'lit/decorators.js';
+import { BaseTab } from './BaseTab.js';
 import { getRandomId } from '@patternfly/pfe-core/functions/random.js';
-import { Logger } from '@patternfly/pfe-core/controllers/logger.js';
 
-import style from './pfe-tab.scss';
-
-function isSlotElement(el?: Node|null): el is HTMLSlotElement {
-  return el instanceof HTMLElement && el.tagName === 'SLOT';
-}
+import style from './BaseTab.scss';
+import pfeStyle from './pfe-tab.scss';
 
 /**
- * @slot - Add the heading for your tab here.
+ * PfeTab
+ *
+ * @csspart button - button element
+ * @csspart icon - span container for the icon
+ * @csspart text - span container for the title text
+ *
+ * @slot icon - We expect the light DOM of `<svg>` or `<pfe-icon>`
+ * @slot - The tab title text
+ *
+ * @fires { TabExpandEvent } tab-expand - when a tab is selected
+ * @fires { CustomEvent<{ selected: 'true'|'false'; tab BaseTab }> }
  */
-@customElement('pfe-tab') @pfelement()
-export class PfeTab extends LitElement {
+@customElement('pfe-tab')
+export class PfeTab extends BaseTab {
   static readonly version = '{{version}}';
 
-  static readonly styles = [style];
-
-  /** If the tab is selected */
-  @observed
-  @property({ reflect: true, attribute: 'aria-selected' })
-    selected: 'true'|'false' = 'false';
+  static readonly styles = [style, pfeStyle];
 
-  /** Connected panel ID */
-  @property({ reflect: true, attribute: 'aria-controls' }) controls?: string;
-
-  /** Variant */
-  @property({ reflect: true }) variant: 'wind'|'earth' = 'wind';
-
-  /**
-   * Sets color theme based on parent context
-   */
-  @colorContextConsumer()
-  @property({ reflect: true }) on?: ColorTheme;
-
-  /** @deprecated `tabIndex` property reflects per spec */
-  get tabindex() {
-    return this.tabIndex;
-  }
-
-  set tabindex(v: number) {
-    this.tabIndex = v;
-  }
-
-  @query('#tab') private _tabItem?: HTMLElement;
-
-  private logger = new Logger(this);
-
-  connectedCallback() {
+  async connectedCallback() {
```

Comment:
as elsewhere, maybe we can use firstUpdated to avoid async connectedCallback. this is a small nit

Reply:
👍